File: protobuf/repeat/pub.js
var zmq = require('zmq')
, protobuf = require('protocol-buffers')
, fs = require('fs')
, messages = protobuf(fs.readFileSync('type.proto'))
, pub = zmq.socket('pub')
, command = {
    "cmd_header":{
        "receiver": "mbtcp",
        "sender":   "restful",        
        "version":  1,
        "tid":      33,                
        "method":   "mbtcp.once.write"
    },
    "mb_tcp_header":{
        "ip": "192.168.1.1", 
        "port": 503,
        "id": 22
    },
    "requests": [ 
        {
            "code":     1,                 
            "register": 2003,             
            "value":    "1025",            
            "type":     "int64",        
            "alias":    "hello_1"  
        }, 
        {
            "code":     2,                 
            "register": 3003,             
            "value":    "2025",            
            "type":     "int32",        
            "alias":    "hello_2"  
        }
    ]
}
, buf = messages.MbTcpMultipleWriteReq.encode(command); // encode

pub.connect("ipc:///tmp/dummy"); // connect to zmq endpoint

setInterval(function() {
    pub.send("mbtcp.once.write", zmq.ZMQ_SNDMORE);
    pub.send(buf);
}, 1000); // emit every 1 seconds

File: protobuf/repeat/pub.go
package main

import (
	_ "encoding/json"
	"fmt"
	proto "github.com/golang/protobuf/proto"
	zmq "github.com/taka-wang/zmq3"
	"time"
)

func main() {
	pub()
}

func pub() {
	sender, _ := zmq.NewSocket(zmq.PUB)
	defer sender.Close()
	sender.Connect("ipc:///tmp/dummy")

	command := &MbTcpMultipleWriteReq{ // named key
		CmdHeader: &CmdHeader{
			Receiver: "core",
			Sender:   "me",
			Version:  1,
			Tid:      32,
			Method:   "mbtcp.once.write",
		},
		MbTcpHeader: &MbTcpHeader{
			Ip:   "192.168.1.1",
			Port: 503,
			Id:   22,
		},
	}

	command.Requests = append(command.Requests,
		&MbWriteRequest{
			Code:     1,
			Register: 2003,
			Value:    "1025",
			Type:     "int64",
			Alias:    "hello",
		},
	)
	command.Requests = append(command.Requests,
		&MbWriteRequest{
			Code:     2,
			Register: 3003,
			Value:    "2025",
			Type:     "int32",
			Alias:    "hello_2",
		},
	)

	cmd, err := proto.Marshal(command)
	if err != nil {
		fmt.Println("marshaling error: ", err)
	}

	for {
		sender.Send("mbtcp.once.write", zmq.SNDMORE)
		sender.Send(string(cmd), 0)
		time.Sleep(time.Duration(1000) * time.Millisecond)
	}
}
